### DISCUSSION TITLE
f-strings in python

### DISCUSSION THREAD

**Discussion by kajuburfi** [helix-editor/helix#2368] [General]
Can it be possible to Auto insert double quotes(or single quotes) when typing an f-string?

Currently, if we type `f"`, it doesn't auto-pair the other side.
And if we again type `"`, it ends up like this `f"""`

Thoughts?

**Comment by ccjoel** [comment]
Hm. I've been running into this. I don't mind preemptively typing `""` (auto-closes) without the `f` prefix, authoring the contents, and then doing some good-ol' helix-fu to add. Example, assume cursor is somewhere inside the string:
```
logger.info("This is my value := {some_var}")
```
helix-fu: `?"<ENTER>if`

Working for me so far, although I guess I should record a macro at this point. Recording to `f` register: 
```
"fQ?"<ENTER>if<ESC>wQ
```
Then, to add `f` prefix anytime where cursor is within the double quotes, invoke the macro: `"fq`.
Now I need to google (or bing-o!) how to persist macros in helix.

**Comment by jerabaul29** [comment]
Curious if anybody has a good "workaround" for this while waiting for something that answers https://github.com/helix-editor/helix/issues/4035 ? :)

Personally I am using the trick of i) entering `"` (which actually auto-writes `""`, then ii) moving to the start and writing the `f`, then iii) writing whatever I want. It works fine enough (this is just 1 extra key), but wonder if there is a better way :) . I guess once https://github.com/helix-editor/helix/pull/9801 lands one could just have a snippet as an alternative? :)
